### DISCUSSION TITLE
[Feature] 请问下如何在其他项目中使用MMClassification？

### DISCUSSION THREAD

**Discussion by David-19940718** [open-mmlab/mmclassification#1114] [General]
### 描述该功能

如题所示，假设我想结合YOLOv5和MMCls这两个工程，要如何在yolov5项目上调用MMCls实现先检测后分类？

### 是否希望自己实现该功能？

- [ ] 我希望自己来实现这一功能，并向 MMClassification 贡献代码！

**Comment by mm-assistant** [comment]
We recommend using English or English & Chinese for issues so that we could have broader discussion.

**Comment by Ezra-Yu** [comment]
直接结合yolov5和mmcls可能非常困难。结合使用mmyolo中的[yolov5](https://github.com/open-mmlab/mmyolo/tree/main/configs/yolov5) 和mmcls是可以的。

你说的结合是 同一个backbone，出来的特征图有两个分支，一个head做分类，一个做yolov5检测。
还是检测好的bbox做分类？

**Reply by David-19940718** [reply]
你好，不是，我不做训练。就是两个独立的模型，我想用yolov5预测完，然后将预测的图片的crop出来送入mmcls推理修正类别。

**Comment by mzr1996** [comment]
请问你是使用的 mmcls 1.x 版本还是 0.x 版本？

**Reply by David-19940718** [reply]
最新版本

**Reply by Ezra-Yu** [reply]
是 `master` 分支， 还是 `1.x` 分支？

运行 ```pip list | grep "mmcls\|mmcv\|torch"``` 查看.

**Reply by David-19940718** [reply]
> 是 `master` 分支， 还是 `1.x` 分支？
> 
> 运行 `pip list | grep "mmcls\|mmcv\|torch"` 查看.

master分支

**Comment by Ezra-Yu** [comment]
使用 [init_model](https://github.com/open-mmlab/mmclassification/blob/a9489f6bd096728eb885e9351f3e65fb78775a7d/mmcls/apis/inference.py#L14) 和 [inference_model](https://github.com/open-mmlab/mmclassification/blob/a9489f6bd096728eb885e9351f3e65fb78775a7d/mmcls/apis/inference.py#L54) 接口；
inference_model 传入crop的图片。注意cfg中[test_pipeline](https://github.com/open-mmlab/mmclassification/blob/a9489f6bd096728eb885e9351f3e65fb78775a7d/configs/_base_/datasets/imagenet_bs32.py#L14) 需要去除Resize 和 CenterCrop
